src/components/Dialog.jsx

```diff
@@ -1,18 +1,60 @@
 import React  from 'react';
+import { CloseIcon } from './Icons.jsx';
 import '../../sass/components/_ciq-dialog.scss';
 
 const Dialog = ({
     open,
     children,
     onContainerClick,
     className,
+    title,
+    setOpen,
+    enableTabular,
+    enableOverlay,
 }) => (
-    <div
-        className={`${className || 'cq-dialog'} ${open ? ' open' : ''}`}
-        onClick={onContainerClick}
-    >
-        {children}
-    </div>
+    <>
+        {enableOverlay ? (
+            <div className={`cq-dialog-overlay ${open ? 'cq-dialog-active' : ''}`}>
+                <div
+                    className={`${className || 'cq-dialog'} ${enableTabular ? 'cq-dialog--tabular' : ''} ${open ? ' open' : ''}`}
+                    onClick={onContainerClick}
+                >
+                    {title ? (
+                        <div className="cq-dialog__head">
+                            <div className="cq-dialog__head--title">{title}</div>
+                            <div className="cq-dialog__head--action">
+                                <CloseIcon
+                                    onClick={() => setOpen(false)}
+                                />
+                            </div>
+                        </div>
+                    ) : ''}
+                    <div className="cq-dialog__body">
+                        {children}
+                    </div>
+                </div>
+            </div>
+        ) : (
+            <div
```

Comment:
@morteza-binary I just add this code, because, we have both menu and dialog, after changing, after we implement all menu to the dialog, we can safely remove this part